ignite/contrib/metrics/average_precision.py

```diff
@@ -1,35 +1,25 @@
-from typing import Callable
+from typing import Callable, Optional, Union
 
 import torch
 
 from ignite.metrics import EpochMetric
 
 
-def average_precision_compute_fn(y_preds: torch.Tensor, y_targets: torch.Tensor) -> float:
```

Comment:
Tricky point here is that `average_precision_compute_fn` is a public method. Removing it will lead to a BC breaking change which is usually unwanted...

Reply:
Any idea to avoid that? and is there any other comments to edit everything and push again?

Reply:
I think you can just keep it where it is. I'll review this PR later.

Reply:
For `AveragePrecision` i will just update the test. Okay i will wait to see the full review

Reply:
I think the best what we can do is to keep this functions where they are and simplify things similar to what we did to CohenKappa:

```python
def average_precision_compute_fn(y_preds: torch.Tensor, y_targets: torch.Tensor) -> float:
     # no need to try/except here
     from sklearn.metrics import average_precision_score

    y_true = y_targets.numpy()
    y_pred = y_preds.numpy()
    return average_precision_score(y_true, y_pred)
```

```python
class AveragePrecision(EpochMetric):

    def __init__(self, ...):

        try:
            from sklearn.metrics import average_precision_score
        except ImportError:
            raise RuntimeError("This contrib module requires sklearn to be installed.")

        super(AveragePrecision, self).__init__(
            average_precision_compute_fn,
            output_transform=output_transform,
            check_compute_fn=check_compute_fn,
            device=device,
        )
```

Reply:
Is this for all metrics or just for `AveragePrecision`?

Reply:
Obviously, for all as otherwise it will be a BC-breaking change, right ?

Reply:
Okay then changing the try/except as provided and add the device arg will be good. Now for the tests, is there something to add/remove?